x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/output/FlushAcknowledgement.java

```diff
@@ -31,39 +32,54 @@
     public static final ParseField LAST_FINALIZED_BUCKET_END = new ParseField("last_finalized_bucket_end");
 
     public static final ConstructingObjectParser<FlushAcknowledgement, Void> PARSER = new ConstructingObjectParser<>(
-            TYPE.getPreferredName(), a -> new FlushAcknowledgement((String) a[0], (Date) a[1]));
+            TYPE.getPreferredName(), a -> new FlushAcknowledgement((String) a[0], (Instant) a[1]));
 
     static {
         PARSER.declareString(ConstructingObjectParser.constructorArg(), ID);
         PARSER.declareField(ConstructingObjectParser.optionalConstructorArg(),
-                p -> TimeUtils.parseTimeField(p, LAST_FINALIZED_BUCKET_END.getPreferredName()),
+                p -> TimeUtils.parseTimeFieldToInstant(p, LAST_FINALIZED_BUCKET_END.getPreferredName()),
                 LAST_FINALIZED_BUCKET_END, ObjectParser.ValueType.VALUE);
     }
 
-    private String id;
-    private Date lastFinalizedBucketEnd;
+    private final String id;
+    private final Instant lastFinalizedBucketEnd;
 
-    public FlushAcknowledgement(String id, Date lastFinalizedBucketEnd) {
+    public FlushAcknowledgement(String id, Instant lastFinalizedBucketEnd) {
```

Comment:
nit: Is the autodetect output an epoch number or a string? If it is a long it seems a bit silly to parse it as an Instant (Date previously) then convert back to a long then back to an Instant.

Reply:
Good point.  It's written as a long.  And it's also a long in the XContent representation too, so the round-tripping tests will also work if it's parsed as a long.  I'll change it to do that.

Reply:
Actually, on trying this while just having one constructor it makes the tests a lot less clear, as they'd all have a number as the second argument.

So I'll introduce another constructor that takes a `Long` and use that in the parser but keep the `Instant` constructor so that tests can use it to make clear that the second argument is a time.